src/Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -32,12 +32,14 @@ public void OnConnection(IFeatureCollection features)
         {
```

Comment:
I was thinking about this more and since it's actually the Kestrel specific interface, we should just change this to the `TransportConnection` base class. That's basically the primitive that Kestrel requires to be implemented in order to plug in a transport. It does mean that the `IConnectionHandler` is coupled to this model but that seems like an ok tradeoff.

Reply:
I'll make this change.